src/relayclient/GSNConfigurator.ts

```diff
@@ -130,14 +148,14 @@ export interface GSNDependencies {
   asyncApprovalData: AsyncDataCallback
   asyncPaymasterData: AsyncDataCallback
   scoreCalculator: AsyncScoreCalculator
-  config: GSNConfig
 }
 
 export function getDependencies (config: GSNConfig, provider?: HttpProvider, overrideDependencies?: Partial<GSNDependencies>): GSNDependencies {
   let contractInteractor = overrideDependencies?.contractInteractor
   if (contractInteractor == null) {
     if (provider != null) {
-      contractInteractor = new ContractInteractor(provider, config)
+      const logger = overrideDependencies?.logger ?? createLogger(config.logLevel, config.customerToken, config.hostOverride, config.userIdOverride)
```

Comment:
why do you create logger twice in this method?

Reply:
Fixed